src/CoreWCF.RabbitMQ/src/CoreWCF/Channels/RabbitMqBinding.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using RabbitMQ.Client;
+
+namespace CoreWCF.Channels
+{
+    public class RabbitMqBinding : Binding
+    {
+        private TextMessageEncodingBindingElement _encoding;
+        private bool _isInitialized;
+        public const long DefaultMaxMessageSize = 8192L;
+
+        /// <summary>
+        /// Creates a new instance of the RabbitMQBinding class initialized
+        /// to use the Protocols.DefaultProtocol. The broker must be set
+        /// before use.
+        /// </summary>
+        public RabbitMqBinding()
+            : this(Protocols.DefaultProtocol)
+        { }
+
+        /// <summary>
+        /// Uses the broker specified by the given hostname and port with
+        /// Protocols.DefaultProtocol.
+        /// </summary>
+        /// <param name="hostname">The hostname of the broker to connect to</param>
+        /// <param name="port">The port of the broker to connect to</param>
+        public RabbitMqBinding(string hostname, int port)
+            : this(hostname, port, Protocols.DefaultProtocol)
```

Comment:
Hostname and port should be encoded in the endpoint url and not set on the binding.